src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -371,6 +375,100 @@ void mutex_init() {
 #undef MUTEX_STORAGE
 #undef MUTEX_STORAGE_NAME
 
+static const int MAX_NAMES = 200;
+static const char* _names[MAX_NAMES] = { nullptr };
+static bool _is_unique[MAX_NAMES] = { false };
```

Comment:
Some explanatory comments would be useful. Why do we need to duplicate all the names? What does the `is_unique` array signify? The way it is used in `name2Id` doesn't make sense to me.

Reply:
The names in the `_names` array are not the same order as the named mutexes created from the `mutex_init()` function. There are mutexes created prior to the `mutex_init()` function. The `_is_unique` is to indicate whether a mutex is a singleton instance (e.g. the ones created from the `mutex_init()` function) or not (e.g. the startThread_lock).
I've added some comment in the above declarations.